packages/discord.js/src/managers/GuildEmojiManager.js

```diff
@@ -142,6 +143,27 @@ class GuildEmojiManager extends BaseGuildEmojiManager {
     }
     return this._add(newData);
   }
+
+  /**
+   * Fetches the author for this emoji
+   * @param {EmojiResolvable} emoji The emoji to fetch the author of
+   * @returns {Promise<User>}
+   */
+  async fetchAuthor(emoji) {
+    emoji = this.resolve(emoji);
+    if (!emoji) throw new TypeError('INVALID_TYPE', 'emoji', 'EmojiResolvable', true);
+    if (emoji.managed) {
+      throw new Error('EMOJI_MANAGED');
+    } else {
+      if (!this.guild.me) throw new Error('GUILD_UNCACHED_ME');
+      if (!this.guild.me.permissions.has(Permissions.FLAGS.MANAGE_EMOJIS_AND_STICKERS)) {
+        throw new Error('MISSING_MANAGE_EMOJIS_AND_STICKERS_PERMISSION', this.guild);
+      }
+    }
```

Comment:
No `else` after `throw`, under the same rule as useless `else` after `return`:
```suggestion
    if (emoji.managed) {
      throw new Error('EMOJI_MANAGED');
    }

    const { me } = this.guild;
    if (!me) throw new Error('GUILD_UNCACHED_ME');
    if (!me.permissions.has(Permissions.FLAGS.MANAGE_EMOJIS_AND_STICKERS)) {
      throw new Error('MISSING_MANAGE_EMOJIS_AND_STICKERS_PERMISSION', this.guild);
    }
```

Also tweaked the code a little so the `guild.me` getter isn't called twice.